### DISCUSSION TITLE
blog/options-pattern-in-aspnet-core/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#47] [General]
# blog/options-pattern-in-aspnet-core/

We will learn about using the Options Pattern in ASP.NET Core to Manage, Bind, and Validate your configurations from appsettings.json. Reading Configuration is a very common requirement that you can have in any kind of application.

https://codewithmukesh.com/blog/options-pattern-in-aspnet-core/

**Comment by dmshved** [comment]
Hello Mr. Mukesh, could you create a post about IConfigureOptions & IPostConfigure options? Your guide was extremelty helpful to achieve what I needed but I also wanted to use IOptions<T> pattern while configuring the JwtBearer in my project. 

The thing is that we can't use the IOptions<T> in the JwtBearer lambda, so we're stuck with 2 solutions: 

1. Use IConfiguration - which is obviously weird as we started to use IOptions<T> pattern
2. Use IConfigureOptions<T> or IPostConfigureOptions<T> to configure the service using our POCO class that was configured using IOptions pattern

And in some cases it can be really hard to understand which one to use. 

Could you provide us some professional explanation please? 

Best regards
Dmytro Shved
